### PR TITLE
matplotlib and basemap require setuptools to run properly together

### LINKED ISSUE #3813: mpl_toolkits.basemap fails to import
I'm having trouble importing mpl_toolkits.basemap. I know this had previously worked, so I don't know where to point the finger. Here's what I see:

```
[lee218@rzwiz2:spack.basemap]$ /nfs/tmp2/lee218/delete/spack.basemap/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/python-2.7.13-gcx7wpxjtt7ujlpbahezikqfz4xn2ul6/bin/python -c 'import mpl_toolkits; print(mpl_toolkits.__path__); import mpl_toolkits.basemap'
['/nfs/tmp2/lee218/delete/spack.basemap/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/python-2.7.13-gcx7wpxjtt7ujlpbahezikqfz4xn2ul6/lib/python2.7/site-packages/matplotlib-2.0.0-py2.7-linux-x86_64.egg/mpl_toolkits']
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named basemap

[lee218@rzwiz2:spack.basemap]$ /nfs/tmp2/lee218/delete/spack.basemap/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/python-3.6.0-ndtiujzzf7p4m5t4k6xergu3f2u674sk/bin/python3 -c 'import mpl_toolkits; print(mpl_toolkits.__path__); import mpl_toolkits.basemap'
['/nfs/tmp2/lee218/delete/spack.basemap/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/python-3.6.0-ndtiujzzf7p4m5t4k6xergu3f2u674sk/lib/python3.6/site-packages/matplotlib-2.0.0-py3.6-linux-x86_64.egg/mpl_toolkits']
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'mpl_toolkits.basemap'
```

With a previous installation I see this:

```
[lee218@rzwiz2:spack.basemap]$ python -c 'import mpl_toolkits; print(mpl_toolkits.__path__); import mpl_toolkits.basemap' ['/collab/usr/gapps/python/build/spack-toss3.1/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/python-2.7.13-gv7myy57t2h6wp44f42b2llku7kiva7z/lib/python2.7/site-packages/matplotlib-1.5.3-py2.7-linux-x86_64.egg/mpl_toolkits', '/collab/usr/gapps/python/build/spack-toss3.1/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/python-2.7.13-gv7myy57t2h6wp44f42b2llku7kiva7z/lib/python2.7/site-packages/mpl_toolkits']
```

Notice that the __path__ in the latter case includes the mpl_toolkits directory where basemap exists. I'm not sure why the old Spack python build was able to find both the matplotlib and basemap mpl_toolkits directories, while the new Spack python build only finds the maplotlib one. Any thoughts @tgamblin @citibeth @adamjstewart @mathstuf?

I found one hack that works around this, which is to add `__path__.append('/nfs/tmp2/lee218/delete/spack.basemap/opt/spack/linux-rhel7-x86_64/gcc-4.8.5/py-basemap-1.0.7-cs5yvwpwtosqpl3t7rhadlccinbmsqvr/lib/python3.6/site-packages/mpl_toolkits')` to .../matplotlib.../mpl_toolkits/__init__.py.

### PR DESCRIPTION
**Pull request by lee218llnl**
Fixes #3813.

matplotlib and basemap require setuptools at runtime in order to allow the mpl_toolkits package to be considered a namespace package and thus look in both matplotlib and basemap's mpl_toolkits directory. This also removes the need for the patch in #1964.

### CODE REVIEW

**Comment by adamjstewart** [comment]
Can you add a short comment above the setuptools dependencies explaining why they are needed at runtime?

**Comment by citibeth** [comment]
This addresses #3813

File: var/spack/repos/builtin/packages/py-basemap/package.py [hunk 1]
```diff
@@ -35,87 +34,11 @@ class PyBasemap(PythonPackage):
 
     version('1.0.7', '48c0557ced9e2c6e440b28b3caff2de8')
 
-    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-basemap/package.py]: Please add a comment for this line that (1) gives a brief explanation and (2) links to https://github.com/LLNL/spack/issues/3813

File: var/spack/repos/builtin/packages/py-matplotlib/package.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ class PyMatplotlib(PythonPackage):
     extends('python', ignore=r'bin/nosetests.*$|bin/pbr$')
 
     # ------ Required dependencies
-    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools', type=('build', 'run'))
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/py-matplotlib/package.py]: Same comment here as on py-basemap

**Comment by adamjstewart** [comment]
@citibeth Can you approve?

**Comment by lee218llnl** [comment]
@citibeth Do the comments in the package that address the "run" dependence look OK to you?

**Comment by adamjstewart** [comment]
Looks like there's a conflict. Can you rebase/merge?

**Comment by lee218llnl** [comment]
@adamjstewart not sure why it considered that a conflict as the "change" in the develop branch was blank.

File: var/spack/repos/builtin/packages/py-basemap/package.py [hunk 3]
```diff
@@ -35,86 +34,15 @@ class PyBasemap(PythonPackage):
 
     version('1.0.7', '48c0557ced9e2c6e440b28b3caff2de8')
 
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-basemap/package.py]: This extra line will cause the flake8 test to fail.

**Comment by lee218llnl** [comment]
@adamjstewart looks like we passed through travis. I think this is ready to be merged unless there's anything else you would like me to address.

**Pull request merged**